src/Cascader.tsx

```diff
@@ -233,7 +246,17 @@ const Cascader = React.forwardRef<CascaderRef, InternalCascaderProps>((props, re
   );
 
   // =========================== Option ===========================
-  const mergedOptions = React.useMemo(() => options || [], [options]);
+
+  // Pass the parent disabled property to the children
+  const initialOptions = (newOptions: DefaultOptionType[] = []) => {
+    newOptions.forEach(newOption => {
+      if (newOption.children) {
+        newOption.children.forEach(appendDisabled(newOption.disabled));
```

Comment:
Use `map` to make sure `option` is not modified.

Reply:
```suggestion
  const initialOptions = (newOptions: DefaultOptionType[] = [], disabled: boolean = false) => {
    return newOptions.map(option => {
      if (disabled || option.disabled) {
        option.disabled = true;
      }

      if (option.children) {
        initialOptions(option.children, option.disabled);
      }

      return option;
    });
  };
```

How about refactoring this code? What do you think ?